Atomic/sign.py

```diff
@@ -0,0 +1,107 @@
+from . import util
+from . import Atomic
+import os
+import tempfile
+from .atomic import AtomicError
+import re
+
+
+class Sign(Atomic):
+    def sign(self):
+        # TODO
+        # Atomic is run as sudo. Should we work around that?
+
+        if self.args.debug:
+            util.write_out(str(self.args))
+
+        # Verify all the images provided are valid
+        self.check_input_validity()
+
+        signature_path = util.get_signature_path(self.atomic_config)
+        use_def_sig_path = True if signature_path == self.args.output else False
+
+        signer = self.args.sign_key
+
+        for sign_image in self.args.images:
+            remote_inspect_info = util.skopeo_inspect("docker://" + sign_image)
+            manifest = util.skopeo_inspect('docker://{}'.format(sign_image), args=['--raw'], return_json=False)
+
+            # Make sure the local image is the same as the image on the
+            # registry
+            manifest_file = tempfile.NamedTemporaryFile(mode="wb", delete=False)
+            manifest_file.write(manifest)
+            manifest_file.close()
+            manifest_hash = str(util.skopeo_manifest_digest(manifest_file.name))
+
+            if use_def_sig_path:
+                _image_path = str(remote_inspect_info['Name'])
+                image_name = os.path.basename(_image_path)
+
+                _sig_path = [signature_path, os.path.dirname(_image_path), "{}@{}".format(image_name, manifest_hash)]
+                sigstore_path = str(os.path.join(*_sig_path))
```

Comment:
In that case I don’t understand why the `dirname`/`basename`/`join` sequence is needed, instead of simpler `"{}/{}@{}".format(signature_path, _image_path, manifest_hash)`.